### PR TITLE
[workflows] combine changeset and lint workflows

### PR DESCRIPTION
**Pull request by taylor-vann**
This PR combines the `changeset` and `lint` workflows into one workflow `changeset-lint`.

These workflows are similar enough to combine :)

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -21,27 +21,21 @@ jobs:
...
-        with:
-          node-version: 16
-          # Note: we do not cache here because we're not running lerna
-          # bootstrap, so the ~/.npm dir won't have all dependencies, and if
-          # this job finishes first then we'll only cache root dependencies.
-
-      - run: npm ci
-      - run: npm run lint
+      - name: NPM install
+        # Enforce that all PRs that change packages need changesets. Changes
```
Comment (justinfagnani) [inline .github/workflows/tests.yml:25]: you can remove this comment

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ name: Tests
 on: [push, pull_request]
 
 jobs:
-  changeset:
+  changeset-lint:
```
Comment (justinfagnani) [inline .github/workflows/tests.yml:6]: I'd be ok just calling this lit and considering checking for a changeset a lint step. @kevinpschaaf ?

Comment (kevinpschaaf) [reply .github/workflows/tests.yml:6]: s/lit/lint
yeah

Comment (taylor-vann) [reply .github/workflows/tests.yml:6]: so just `lint` ? confirming

Comment (taylor-vann) [reply .github/workflows/tests.yml]: done!

File: .github/workflows/tests.yml (hunk 1 above)
Comment (taylor-vann) [reply .github/workflows/tests.yml]: sounds good! done!

**Review by justinfagnani — APPROVED** [review]

**Pull request merged**